graphium/nn/architectures/global_architectures.py

```diff
@@ -1341,9 +1352,28 @@ def forward(self, g: Batch) -> Tensor:
                 e = self.pre_nn_edges.forward(e)
             g["edge_feat"] = e
 
-        # Run the graph neural network
-        g = self.gnn.forward(g)
+        # Apologies for the similar names here
+        extras_to_return = []
+
+        if extra_return_names:
+            # Run the graph neural network
+            g, gnn_extras_to_return = self.gnn.forward(g, extra_return_names=extra_return_names)
+            extras_to_return.update(gnn_extras_to_return)
+
+            if "pre_task_heads" in extra_returns:
+                extras_to_return.update({"pre_task_heads": g})
 
+            if self.task_heads is None:
+                return g, extras_to_return
+
+            final_output, task_head_extras_to_return = self.task_heads.forward(
+                g, extra_return_names=extra_return_names
+            )
+            extras_to_return.update(task_head_extras_to_return)
+
+            return final_output, extras_to_return
+
+        # Keep original code if no extra_return_names
```

Comment:
Maybe I'm missing something obvious, in which case apologies. 
Aren't you missing the original forward pass in the case the `extra_return_names` is missing?
```suggestion
        # Keep original code if no extra_return_names
        else:
             g = self.gnn.forward(g)
```